shared/templates/firefox_policy-setting/template.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import print_function
+import re
+import json
+
+
+def normalize_policy_path(path_string):
+    """Generate a normalized path array from yaml.
+       Paths are space-separated key names.
+       All firefox policies are under the policies key.
+    """
+    path = []
+    if len(path_string.strip()) > 0:
+        path = re.split(" ", path_string)
+    if 'policies' not in path:
+        path = ['policies'] + path
+    return path
+
+
+def make_json_boolean(instr, oval_format=False):
+    if instr.lower() in ["false", "true"]:
+        if oval_format:
+            return r"[{0}{1}]{2}".format(instr[0].upper(), instr[0].lower(), instr[1:])
+        else:
+            return instr.lower() == "true"
+    if oval_format:
+        return "\"{0}\"".format(instr)
+    return "\'{0}\'".format(instr)
+
+
+def build_oval_search_regex_for_json_parameter(_path, parameter, value):
+    parameter = r'"{0}"[\s]*:[\s]*{1},?'.format(parameter, make_json_boolean(value,
+                                                                             oval_format=True))
+    _result = parameter
+    for p in reversed(_path):
+        if p == "policies":
+            _result = ''.join([r'"{0}"[\s]*:[\s]*'.format(p),
+                               r"\{.*[\s]*",
+                               _result,
+                               r"[^}]*\}"])
+        else:
+            _result = ''.join([r'"{0}"[\s]*:[\s]*'.format(p),
+                               r"(\{?(\{[^}]*\}?)*|[^}]*)[\s]*",
+                               _result,
+                               r"[^}]*\}"])
+    return _result
+
+
+def build_python_json_notation(_path, tier=0):
```

Comment:
I have no idea what this function is supposed to do. What is tier? And what is python-specific about the functionality?

Reply:
When generating the inlined python script for remediation, I need to have the call notation written generated (where it's used in the template as `path_python`). Tier is the distance from the top of the document.
Adding a docstring now.